frontend/src/reducers/usersReducer.tsx

```diff
@@ -1,16 +1,39 @@
 import * as actionTypes from 'constants/actionTypes';
 import { IPagedItems, IUser } from 'interfaces';
-import { IStoreUsersAction, IUpdateUserAction } from 'actions/adminActions';
+import {
+  IStoreUsersAction,
+  IUpdateUserAction,
+  IFilterUsersAction,
+  IUsersPageSizeAction,
+  IUsersSort,
+  ISortUsersAction,
+} from 'actions/adminActions';
+
+export const MAX_USERS_PER_PAGE: number = 10;
 
 export interface IUsersState {
   pagedUsers: IPagedItems<IUser>;
+  rowsPerPage: number;
+  filterList: string[][];
+  sort: IUsersSort;
 }
 
 const initialState: IUsersState = {
-  pagedUsers: { page: 0, total: 0, quantity: 0, items: [] },
+  pagedUsers: { page: 1, pageIndex: 0, total: 0, quantity: 0, items: [] },
+  rowsPerPage: MAX_USERS_PER_PAGE,
+  filterList: [[], [], [], [], [], [], [], [], [], []],
```

Comment:
could this have been implemented as a dictionary instead (an object of keys where each key is an array)? Even if we needed to create a translation function between the dictionary and this format I think that would be my preference.

Reply:
I am a little hesitant to invest any more time to clean it 100% because we still have to refactor after deciding on which grid system

Reply:
agree that further work should be pushed to the refactor